apollo-router/src/main.rs

```diff
@@ -134,22 +134,44 @@ async fn rt_main() -> Result<()> {
         })
         .unwrap_or_else(|| ConfigurationKind::Instance(Configuration::builder().build().boxed()));
 
-    let schema = if let Ok(apollo_key) = std::env::var("APOLLO_KEY") {
-        let apollo_graph_ref = std::env::var("APOLLO_GRAPH_REF")
-            .expect("cannot set up Uplink schema download if  the APOLLO_GRAPH_REF environment variable is not set");
+    let schema = match (opt.supergraph_path, std::env::var("APOLLO_KEY")) {
+        (Some(supergraph_path), _) => {
+            let supergraph_path = if supergraph_path.is_relative() {
+                current_directory.join(supergraph_path)
+            } else {
+                supergraph_path
+            };
+            SchemaKind::File {
+                path: supergraph_path,
+                watch: opt.watch,
+                delay: None,
+            }
+        }
+        (None, Ok(apollo_key)) => {
+            let apollo_graph_ref = std::env::var("APOLLO_GRAPH_REF")
+            .map_err(|_| anyhow!("cannot fetch the supergraph from Apollo Studio without setting the APOLLO_GRAPH_REF environment variable"))?;
 
-        SchemaKind::Registry {
-            apollo_key,
-            apollo_graph_ref,
+            SchemaKind::Registry {
+                apollo_key,
+                apollo_graph_ref,
+            }
         }
-    } else {
-        ensure!(
-            opt.supergraph_path.is_some(),
-            r#"
+        _ => {
+            return Err(anyhow!(
+                r#"
     💫 Apollo Router requires a supergraph to be set using '--supergraph':
 
         $ ./router --supergraph <file>`
 
+        Alternatively, to retrieve the supergraph from Apollo Studio, set the APOLLO_KEY
```

Comment:
💫 Apollo Router requires a supergraph which may be specified using either '--supergraph':
        `$ ./router --supergraph <file>`
        or, to retrieve the supergraph from Apollo Studio, set the APOLLO_KEY

Reply:
Ah, I see what you're capturing there and completely agree with it being clarity-adding.

I'm on my phone right now, but suggest land this and we get help from DX on word-smithing this a bit.  I could even imagine that we want to hoist managed federation to the top to some degree.

Reply:
I still don't love my text to be honest. I think we may need some additional explanatory text which shows that "if both --supergraph and APOLLO_KEY are set, --supergraph will be used" (or something like this). How about this?:


Apollo Router requires a supergraph. If --supergraph is specified then the supergraph is read from the specified file. If not, then, as long as APOLLO_KEY and APOLLO_GRAPH_REF environment variables are set, the supergraph will be retrieved from Apollo Studio. Note: --supergraph always takes precedence.

Reply:
Marking this as resolved as it's now tracked via https://github.com/apollographql/router/issues/549 to track it.